substrate/client/src/in_mem.rs

```diff
@@ -108,16 +103,17 @@ impl<Block: BlockT> Blockchain<Block> {
 	}
 
 	/// Create new in-memory blockchain storage.
-	pub fn new() -> Self {
+	pub fn new() -> Blockchain<Block> {
+		let storage = Arc::new(RwLock::new(
+			BlockchainStorage {
+				blocks: HashMap::new(),
+				hashes: HashMap::new(),
+				best_hash: Default::default(),
+				best_number: Zero::zero(),
+				genesis_hash: Default::default(),
+			}));
 		Blockchain {
-			storage: RwLock::new(
-				BlockchainStorage {
-					blocks: HashMap::new(),
-					hashes: HashMap::new(),
-					best_hash: Default::default(),
-					best_number: Zero::zero(),
-					genesis_hash: Default::default(),
-				})
+			storage: storage.clone(),
```

Comment:
Is this clone really needed?

Reply:
Not in this PR :) It is actually from the next PR (https://github.com/paritytech/polkadot/pull/251/files#diff-8f118ee188780f9183b6af320625cfb9R106)